astroquery/heasarc/core.py

```diff
@@ -1,92 +1,247 @@
-from ..query import BaseQuery
-from ..utils import commons
-from ..utils import async_to_sync
+from astropy.utils.decorators import deprecated, deprecated_renamed_argument
+
+import pyvo
+
+from astroquery import log
+from ..query import BaseQuery, BaseVOQuery
+from ..utils import commons, async_to_sync, parse_coordinates
 from ..exceptions import InvalidQueryError, NoResultsWarning
 from . import conf
 
-__all__ = ['Heasarc', 'HeasarcClass']
 
+@async_to_sync
+class HeasarcClass(BaseVOQuery, BaseQuery):
+    """Class for accessing HEASARC data using XAMIN.
 
-def Table_read(*args, **kwargs):
-    if commons.ASTROPY_LT_5_1:
-        return Table.read(*args, **kwargs)
-    else:
-        return Table.read(*args, **kwargs, unit_parse_strict='silent')
+    Example Usage:
+    ...
 
+    """
 
-@async_to_sync
-class HeasarcClass(BaseQuery):
+    # we can move url to Config later
+    VO_URL = conf.VO_URL
+    TAR_URL = conf.TAR_URL
+    S3_BUCKET = conf.S3_BUCKET
+    timeout = conf.timeout
+
+    def __init__(self):
+        """Initialize some basic useful variables"""
+        super().__init__()
+        self._tap = None
+        self._datalink = None
+        self._meta_info = None
+        self._session = None
+
+    @property
+    def tap(self):
+        """TAP service"""
+        if self._tap is None:
+            self._tap = pyvo.dal.TAPService(
+                f'{self.VO_URL}/tap', session=self._session
+            )
+            self._session = self._tap._session
+        return self._tap
+
+    @property
+    def _meta(self):
+        """Queries and holds meta-information about the tables.
+
+        This is a table that holds useful information such as
+        the list of default columns per table, the reasonable default
+        search radius per table that is appropriate for a mission etc.
+        Instead of making a server call for each table for that type information,
+        we do a single one and then post-process the resulting table.
+
+        These are not meant to be used directly by the user.
+        """
+        if self._meta_info is None:
+            query = (
+                "SELECT split_part(name, '.', 1) AS table, "
+                "split_part(name, '.', 2) AS par, "
+                "CAST(value AS DECIMAL) AS value "
+                "FROM metainfo "
+                "WHERE (type = 'parameter' and relation = 'order') "
+                "OR relation LIKE 'defaultSearchRadius' "
+                "ORDER BY value"
+            )
+            self._meta_info = self.query_tap(query).to_table()
+            self._meta_info['value'] = np.array(
+                self._meta_info['value'], np.float32)
+            self._meta_info = self._meta_info[self._meta_info['value'] > 0]
+        return self._meta_info
+
+    def _get_default_cols(self, table_name):
+        """Get a list of default columns for a table
 
-    """
-    HEASARC query class.
+        Parameters
+        ----------
+        table_name: str
```

Comment:
Extreme nitpick, but there should be a space before `:` in parameter lists, otherwise sphinx won't render it properly. Here and everywhere below